### ISSUE TITLE
Segmentation fault in wazuh-modulesd in a ppc64le host

### ISSUE THREAD

**Issue by BraulioV** [wazuh/wazuh#3586]
|Wazuh version|Component|Install type|Install method|Platform|
|---|---|---|---|---|
| 3.9.2-3925 | `wazuh-modulesd` | Manager | Packages | Ubuntu 18.04 ppc64le |

Hi team,

In Ubuntu 18.04 ppc64le hosts, `wazuh-modulesd` crashes with a segmentation fault: 
```
2019/06/27 12:33:42 wazuh-modulesd[18559] wdb.c:155 at wdb_open_agent(): DEBUG: No SQLite database found for agent 'localhost', creating.
2019/06/27 12:33:42 wazuh-modulesd[18559] wdb_agent.c:312 at wdb_create_agent_db(): DEBUG: Profile database not found, creating.
2019/06/27 12:33:42 wazuh-modulesd:database[18559] wm_database.c:1424 at wm_inotify_start(): DEBUG: Waiting for event notification...
2019/06/27 12:33:43 wazuh-modulesd:syscollector[18559] syscollector_common.c:76 at wm_sys_main(): INFO: Starting evaluation.
2019/06/27 12:33:43 wazuh-modulesd:syscollector[18559] syscollector_linux.c:981 at sys_network_linux(): DEBUG: Starting network inventory.
2019/06/27 12:33:43 wazuh-modulesd:syscollector[18559] syscollector_linux.c:1017 at sys_network_linux(): DEBUG: sys_network_linux() sending '{"type":"network","ID":1632413537,"timestamp":"2019/06/27 12:33:43","iface":{"name":"eth0","type":"ethernet","state":"up","MAC":"fa:16:3e:b5:bb:92","tx_packets":436260,"rx_packets":905007,"tx_bytes":94975087,"rx_bytes":2263189974,"tx_errors":0,"rx_errors":0,"tx_dropped":0,"rx_dropped":20618,"MTU":1500,"IPv4":{"address":["140.211.169.159"],"netmask":["255.255.255.192"],"broadcast":["140.211.169.191"],"metric":100,"gateway":"140.211.169.129","DHCP":"enabled"},"IPv6":{"address":["fe80::f816:3eff:feb5:bb92"],"netmask":["ffff:ffff:ffff:ffff::"],"DHCP":"enabled"}}}'
2019/06/27 12:33:43 wazuh-modulesd:syscollector[18559] syscollector_linux.c:1017 at sys_network_linux(): DEBUG: sys_network_linux() sending '{"type":"network","ID":1632413537,"timestamp":"2019/06/27 12:33:43","iface":{"name":"docker0","type":"ethernet","state":"down","MAC":"02:42:c8:b0:2f:7a","tx_packets":151691,"rx_packets":82038,"tx_bytes":1427095740,"rx_bytes":4997904,"tx_errors":0,"rx_errors":0,"tx_dropped":0,"rx_dropped":0,"MTU":1500,"IPv4":{"address":["172.17.0.1"],"netmask":["255.255.0.0"],"broadcast":["172.17.255.255"],"gateway":"unknown","DHCP":"enabled"},"IPv6":{"address":["fe80::42:c8ff:feb0:2f7a"],"netmask":["ffff:ffff:ffff:ffff::"],"DHCP":"enabled"}}}'
2019/06/27 12:33:43 wazuh-modulesd:syscollector[18559] syscollector_linux.c:1037 at sys_network_linux(): DEBUG: sys_network_linux() sending '{"type":"network_end","ID":1632413537,"timestamp":"2019/06/27 12:33:43"}'
2019/06/27 12:33:43 wazuh-modulesd:syscollector[18559] syscollector_linux.c:912 at sys_os_unix(): DEBUG: Starting Operating System inventory.
2019/06/27 12:33:43 wazuh-modulesd:syscollector[18559] syscollector_linux.c:926 at sys_os_unix(): DEBUG: sys_os_unix() sending '{"type":"OS","ID":2023028601,"timestamp":"2019/06/27 12:33:43","inventory":{"os_name":"Ubuntu","os_major":"18","os_minor":"04","os_version":"18.04.2 LTS (Bionic Beaver)","os_codename":"Bionic Beaver","os_platform":"ubuntu","sysname":"Linux","hostname":"wazuh2","release":"4.15.0-45-generic","version":"#48-Ubuntu SMP Tue Jan 29 16:27:02 UTC 2019","architecture":"ppc64le"}}'
2019/06/27 12:33:43 wazuh-modulesd:syscollector[18559] syscollector_linux.c:848 at sys_hw_linux(): DEBUG: Starting Hardware inventory.
Segmentation fault (core dumped)
```

**Comment by chemamartinez** [comment]
Hi @BraulioV,

Investigating deeper the issue, the hardware scan looks for the CPU name at the file `/proc/cpuinfo`, particularly, the tag `model name` which looks like the next one in my Ubuntu 18:

```
# cat /proc/cpuinfo
...
model name	: Intel(R) Core(TM) i7-7660U CPU @ 2.50GHz
```

That tag is missing in Ubuntu 18 ppc64le:

```
# cat /proc/cpuinfo
processor	: 0
cpu		: POWER8 (architected), altivec supported
clock		: 3425.000000MHz
revision	: 2.1 (pvr 004b 0201)

timebase	: 512000000
platform	: pSeries
model		: IBM pSeries (emulated by qemu)
machine		: CHRP IBM pSeries (emulated by qemu)
MMU		: Hash
```

So, when reading that info, the variable `sys_info->cpu_name` is `NULL` at this line: https://github.com/wazuh/wazuh/blob/bc2ba4c147b21b425d29f89851f120b9554ac949/src/wazuh_modules/syscollector/syscollector_linux.c#L866

In `w_strtrim()`, a `NULL` input makes the module crash here: https://github.com/wazuh/wazuh/blob/bc2ba4c147b21b425d29f89851f120b9554ac949/src/shared/string_op.c#L162

**Comment by chemamartinez** [comment]
Merged #3624. Thanks for reporting it @BraulioV.
